### PR TITLE
Implemented API HTTP requests for Lobby list (GET/POST) and detail view (GET/PATCH/DELETE)

### LINKED ISSUE #52: Create API endpoint: GET/POST list of Lobbies (URL: /api/lobbies/)
**Goal**

Set up an endpoint that retrieves information about a list of ChiGame lobbies from API, serialized in JSON. This will be done at url "api/lobbies/", and can be accessed by ChiGame users as well as third-party websites. The lobbies get request would include how many players, what game is being played (including whether its tournament or match), and availability (private or public lobby, etc.) for all lobbies.

ENDPOINT: `/api/lobbies`: Supports `GET` and `POST` requests in order to get a list of lobbies (and the corresponding information) as well as be able to create new lobbies using the API

### PR DESCRIPTION
**Pull request by abrahmasandra**
Implemented the list view for Lobby objects, as well as the detail view under the endpoints `/api/lobbies` and `/api/lobbies/<int:pk>` respectively.

ENDPOINTS:

`/api/lobbies/`: This supports both GET and POST requests. GET will return a list of all the lobby objects in the database using Django REST Framework. POST allows users to create new lobbies

`/api/lobbies/<int:LobbyID>`: This supports GET/PATCH/DELETE requests through Django REST framework. GET returns a detail view for a specific lobby, which PATCH allows a user to edit the contents of the lobby, and DELETE allows user to delete a certain lobby.

TASK COMPLETED:

When the endpoints are finished being implemented -- meaning all the functionality described above is complete.

### CODE REVIEW

**Comment by abrahmasandra** [comment]
Created the Lobby list and detail URL endpoints and views, along with a `LobbySerializer` class for the API app within Chigame.

**Review by cuyakonwu — APPROVED** [review]
Code seems appropriately implemented, will need need further communication from lobbies team moving forward

File: src/chigame/api/serializers.py [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+            "id",
+            "name",
+            "game",
+            "game_mod_status",
+            "created_by",
+            "members",
+            "min_players",
+            "max_players",
+            "time_constraint",
+            "lobby_created",
```
Comment (majorsylvie) [inline src/chigame/api/serializers.py:25]: Very good generalizable formatting for listing the specific fields.
I recommend you do line separation like this in general when you have lists of items.

**Review by majorsylvie — CHANGES_REQUESTED** [review]
There are some simpler merge conflicts to be resolved.

Other than that I can succesfully view list and detail, create, delete, and both PUT and PATCH individual models.

Working as expected, resolve the merge conflicts and you'll be good to go.

File: src/chigame/api/serializers.py (hunk 1 above)
Comment (abrahmasandra) [reply src/chigame/api/serializers.py:25]: I merged the `dev` branch into `apis/lobby-endpoints` so all of the merge conflicts should be fixed.

**Review by majorsylvie — APPROVED** [review]
Merge conflicts handled and pre-commit hooks passed, this is good to go! Thank you for addressing feedback.

**Comment by abrahmasandra** [comment]
Closing statement: Added two endpoints `/api/lobbies` and `api/lobbies/<int:pk>`. The former gets a list view of Lobbies and support GET and POST requests. The latter gets the detail view of a specific lobby using the lobby's id and supports GET, PATCH, and DELETE requests.

**Comment by majorsylvie** [comment]
Issue Score: Excellent

Comments:
Great job! 
I genuinely really appreciate the format you use when describing what endpoint were made. Directly listing `endpoint : available requests : details...` is a simple and straightforward way to clearly communicate the most important information for work like this.

Small note, your title is a little off, because you say you implemented the `HTTP requests`, when what you implemented were the `Django Rest Framework Endpoints` which support the requests that you later put in parenthesis `ie: (GET/POST)`

Great work, keep moving forward! We're almost done with the quarter.

**Pull request merged**